tests/ship_client.cpp

```diff
@@ -0,0 +1,143 @@
+#include <eosio/abi.hpp>
+#include <eosio/from_json.hpp>
+#include <eosio/convert.hpp>
+
+#include <rapidjson/document.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/prettywriter.h>
+
+#include <boost/asio.hpp>
+#include <boost/beast.hpp>
+#include <boost/program_options.hpp>
+
+#include <iostream>
+#include <string>
+
+namespace bpo = boost::program_options;
+
+int main(int argc, char* argv[]) {
+   boost::asio::io_context ctx;
+   boost::asio::ip::tcp::resolver resolver(ctx);
+   boost::beast::websocket::stream<boost::asio::ip::tcp::socket> stream(ctx);
+   eosio::abi abi;
+
+   bpo::options_description cli("ship_client command line options");
+   bool help = false;
+   std::string socket_address = "127.0.0.1:8080";
+   uint32_t num_requests = 1;
+
+   cli.add_options()
+      ("help,h", bpo::bool_switch(&help)->default_value(false), "Print this help message and exit.")
+      ("socket-address,a", bpo::value<std::string>(&socket_address)->default_value(socket_address), "Websocket address and port.")
+      ("num-requests,n", bpo::value<uint32_t>(&num_requests)->default_value(num_requests), "number of requests to make");
+   bpo::variables_map varmap;
+   bpo::store(bpo::parse_command_line(argc, argv, cli), varmap);
+   bpo::notify(varmap);
+
+   if(help) {
+      cli.print(std::cout);
+      return 0;
+   }
+
+   std::string::size_type colon = socket_address.find(':');
+   std::string statehistory_server = socket_address.substr(0, colon);
+   std::string statehistory_port = socket_address.substr(colon+1);
```

Comment:
Possible `string::npos + 1` here if no `:` in `-a` which will be `0` and return an empty string, but seems a better error could be provided if `colon` is checked for `npos`.